data/tutorials/guides/1wf_06_database.md

```diff
@@ -0,0 +1,328 @@
+is more secure to use `with_connection` which garantee the closure of the database even in the
+presence of an error.
+
+Then the following template should be used:
+
+```ocaml
+let () =
+  match Lwt_main.run @@
+    Caqti_lwt.with_connection(Uri.of_string "sqlite3:essai.sqlite")
+      (function cnx ->
+         ...
+
+         Lwt.return (Result.Ok ())
+      )
+  with
+  | Result.Ok () -> ...
+  | Result.Error err -> ...
+```
+
+The following URIs are supported:
+
+- `"sqlite3:filename"` (using sqlite3 library),
+- `"postgresql://user:passwd@host/database"` (using postgresql library),
+- `"pgx://user:passwd@host/database"` (using pgx library),
+- `"mariadb://user:passwd@host/database"` (using mariadb library),
+
+
+## Declaring and using queries
+
+### Default queries
+
+The declaration of Caqti queries is rather complex. Hopefully, the `ppx_rapper` preprocessor
+permits an easy composition of such queries.
+
+The idea is to type something like this:
+
+```ocaml
+let query = [%rapper get_many {sql| SELECT @int{id}, @string?{value} FROM a
+                                    WHERE category = %string{category} |sql} ]  
+```
+
+`%type{name}` is used to indicate parameters which will be provided when calling the query.
+`@type{name}` is uses to indicate values (column names) which will be returned by the query.
+
+Types may be `int`, `int32`, `int64`, `string`, `octets`, `float`, `bool`, `pdate`,
+`ptime_span` and `cdate` and `ctime` from `caqti-type-calendar`. A `?` like in this
+example is used to return option values: `None` if the SQL value is `NULL`, `Some v`.
+
+See [ppx_raper reference](https://github.com/roddyyaga/ppx_rapper) for defining custom types.
+
+In order to execute the query, we have to type:
+
+```ocaml
+let* result = query ~category:"a category" cnx
+```
+
+With every input parameters as a labelled argument. If the query has no parameter (no
+`%` values), `let* result = query () cnx` should be typed.
+
+There are different types of queries. A `get_many` query (like the example) returns a list of
+rows. `get_one` just returns the unique row, `get_opt` returns an option value (`None` or `Some`
+depending of the existance of a queried row), `execute` returns nothing `()`.
+
+By default rows are presented as tuples.
+
+### ppx_rapper options
+
+Depending of the needs, a `ppx_rapper` query can be written as
+
+```ocaml
+  [%rapper get_many {sql|...|sql} options]
+```
+
+Where supported options are:
+
+- `record_in` : instead of calling the query with one argument per parameter, a single
+argument is provided. It must be a record whose the fields correspond to the parameters.
+- `record_out` : instead of outputing row(s) as tuple(s), they are presented as record(s).
+- `function_out` : the query is called with a function as its first argument. This function
+is called for each row with every values (column names) as a labelled arguments. The return value of the query
```

Comment:
```suggestion
is called for each row with every value (column names) as a labelled argument. The return value of the query
```
This either needs to be
"every value" or just "values" 
as well as:
"a labelled argument" or "labelled arguments" (without the "a")